src/components/StatsInnerList.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from "react";
+import { LibraryData } from "../interfaces";
+import { getPercentage } from "../utils/sharedFunctions";
+
+export interface StatsInnerListProps {
+  data: any;
+  styled: boolean;
+  stagesToShow?: {[ key: string ]: boolean};
+  showGeographicInfo?: boolean;
+}
+
+export default class StatsInnerList extends React.Component<StatsInnerListProps, {}> {
+  makeCategoryBar(category: string): JSX.Element[] {
+    let catName = category.replace(category[0], category[0].toUpperCase());
+    let length = this.props.data[category].length;
+    let allLengths = Object.values(this.props.data).map(x => (x as any).length);
+    return [
+      <span key={catName}>{catName}: {length}</span>,
+      <span key={`${catName}-${length}`}> ({getPercentage(length, allLengths, true)})</span>
+    ];
+  }
+
+  makeLibraryNameList(category: string): JSX.Element[] {
+    let libraries = this.props.data[category];
+    return libraries.map((l) => {
+      return (
+        <li className="inner-stats-item" key={l.uuid}><p>{l.basic_info.name}{this.getGeographicInfo(l)}</p></li>
+      );
+    });
+  }
+
+  makeLi(category: string): JSX.Element {
+    let hasStyles = this.props.styled;
+    return (
+      <li key={category} className={hasStyles ? "stats-category" : ""}>
+        <section className={hasStyles ? "stats-category-name" : ""}>
+          { this.makeCategoryBar(category) }
+        </section>
+        { (!this.props.stagesToShow || this.props.stagesToShow[category]) &&
+          <ul className={`${hasStyles ? "stats-category-list " : ""}`}>
+            {this.makeLibraryNameList(category)}
+          </ul>
+        }
+      </li>
+    );
+  }
+
+  render() {
+    let list = this.props.data && Object.keys(this.props.data).map(category => {
+      return this.makeLi(category);
+    });
+    return <ul className="stats-inner-list">{list}</ul> || "null";
+  }
+
+  getGeographicInfo(library: LibraryData): string {
```

Comment:
As mentioned offline, this seems to return digits instead of the state information.

Reply:
As also mentioned offline: I can't replicate this :(